lib/src/commit_builder.rs

```diff
@@ -28,10 +26,21 @@ pub struct CommitBuilder<'repo, 'settings> {
     rewrite_source: Option<Commit>,
 }
 
+#[cfg(test)]
 /// TODO: This function is still used in some tests, but shouldn't be. Its
-/// output is not reproducible. Its replacement is available in a follow-up commit.
+/// output is not reproducible.
 pub fn new_change_id_deprecated() -> ChangeId {
-    ChangeId::from_bytes(Uuid::new_v4().as_bytes())
+    ChangeId::from_bytes(uuid::Uuid::new_v4().as_bytes())
+}
+
+pub fn new_change_id(settings: &UserSettings) -> ChangeId {
```

Comment:
I feel like it would be simpler to move this method onto `UserSettings`. Then we don't even need to expose the `random_gen()` function (not until we have another use for it anyway).

Reply:
Sure, if you prefer. I was trying to limit the amount of stuff not related to settings in there.

Reply:
I think we settled on the opposite plan: a reference to the RNG going into CommitBuilder.